src/dom/dom.js

```diff
@@ -3104,53 +3374,79 @@ class MediaElement extends p5.Element {
   }
 
   /**
-   * If no arguments are given, returns the current time of the element.
-   * If an argument is given the current time of the element is set to it.
+   * Manages the media element's playback time. Calling `myMedia.time()`
+   * returns the number of seconds the audio/video has played. Time resets to
+   * 0 when the looping media restarts.
+   *
+   * The parameter, `time`, is optional. It's a number that specifies the
+   * time, in seconds, to jump to when playback begins.
    *
    * @method time
-   * @return {Number} current time (in seconds)
+   * @return {Number} current time (in seconds).
    *
    * @example
-   * <div><code>
-   * let ele;
-   * let beginning = true;
+   * <div>
+   * <code>
+   * let dragon;
+   *
    * function setup() {
-   *   //p5.MediaElement objects are usually created
-   *   //by calling the createAudio(), createVideo(),
-   *   //and createCapture() functions.
-   *
-   *   //In this example we create
-   *   //a new p5.MediaElement via createAudio().
-   *   ele = createAudio('assets/lucky_dragons.mp3');
-   *   background(250);
+   *   // Create a p5.MediaElement using createAudio().
+   *   dragon = createAudio('assets/lucky_dragons.mp3');
+   *   // Show the default media controls.
+   *   dragon.showControls();
+   *
+   *   describe('The text "S seconds" on a gray square with media controls beneath it. The number "S" decreases as the song plays.');
+   * }
+   *
+   * function draw() {
+   *   background(200);
+   *
+   *   // Display the current time.
+   *   let s = dragon.time();
+   *   // Round s to 1 decimal place
+   *   // for display.
+   *   s = round(s, 1);
    *   textAlign(CENTER);
-   *   text('start at beginning', width / 2, height / 2);
+   *   text(`${s} seconds`, 50, 50);
    * }
+   * </code>
+   * </div>
    *
-   * // this function fires with click anywhere
-   * function mousePressed() {
-   *   if (beginning === true) {
-   *     // here we start the sound at the beginning
-   *     // time(0) is not necessary here
-   *     // as this produces the same result as
-   *     // play()
-   *     ele.play().time(0);
-   *     background(200);
-   *     text('jump 2 sec in', width / 2, height / 2);
-   *     beginning = false;
-   *   } else {
-   *     // here we jump 2 seconds into the sound
-   *     ele.play().time(2);
-   *     background(250);
-   *     text('start at beginning', width / 2, height / 2);
-   *     beginning = true;
-   *   }
+   * <div>
+   * <code>
+   * let dragon;
+   *
+   * function setup() {
+   *   // Create a p5.MediaElement using createAudio().
+   *   dragon = createAudio('assets/lucky_dragons.mp3');
+   *   // Show the default media controls.
+   *   dragon.showControls();
+   *
+   *   // Jump to a random starting time.
+   *   let d = dragon.duration();
```

Comment:
oh looks like the tests are failing on this one. It seems like `duration()` is NaN on first load. You might have to do this code block in the loaded callback for `createAudio`?

Reply:
Good catch. Starting at a random time isn't essential, so now it just jumps to 2 seconds.